AddNewPod.md

```diff
@@ -0,0 +1,41 @@
+# Adding a New Firebase CocoaPod
+
+## Introduction
+
+The Firebase build is driven by the contents of a podspec. It is helpful to
+use an existing podspec as a template when starting a new pod.
+
+## Podspec attributes
+
+See the [Podspec Syntax Reference](https://guides.cocoapods.org/syntax/podspec.html) for
+detailed instructions. Some Firebase specific guidance below:
+
+* `s.deployment_target` - Ideally should include ios, osx, and tvos. See
+[FirebaseCore.podspec](FirebaseCore.podspec) for the current Firebase minimum version settings.
+
+* `s.static_framework` - By default, Firebase pods should be static frameworks.
+
+* `s.dependency` - Dependencies on other Firebase pods should allow minor version updates - 
+like `s.dependency 'FirebaseCore', '~> 6.0'`
+
+* `s.pod_target_xcconfig` - Add any specific build settings. For portability, any Firebase
+pod with other Firebase dependencies should build for c99 -
+`'GCC_C_LANGUAGE_STANDARD' => 'c99'`. The pod's version should be passed in as a #define
+for FIRComponent registration. See examples of setting `GCC_PREPROCESSOR_DEFINITIONS`.
+
+* `s.test_spec` should be used for defining all unit and integration test suites.
+
+
+## Directory Structure
+
+The Firebase library `Foo` should be defined in `FirebaseFoo.podspec`. All of its
+contents should be in the `FirebaseFoo` directory.
+
+* FirebaseFoo/Source - All source. Directory structure is up to the library owner.
```

Comment:
How does this work for Firestore, where we have multiple logical entities underneath?

Also, we have embedded third-party code?

Firestore seems like it would benefit from having subpackages. Is that allowed in this structure?

Reply:
Subpackages is still an area of investigation with SPM. Currently, SPM does not even support more than one package per repo.  It looks like it is designed to support it, but we may need to the implementation. Once we dig deeper, we can get a better sense of how to handle subpackages.  

I updated the doc for `third_party`. Thanks.